utils/helper_cloud.sh

```diff
@@ -0,0 +1,697 @@
+  SERVICE_ACCOUNT_ID=$(cloud_create_service_account $SERVICE_NAME)
+  echo "SERVICE_ACCOUNT_ID: $SERVICE_ACCOUNT_ID, SERVICE_NAME: $SERVICE_NAME"
+
+  ENVIRONMENT_NAME="demo-env-$SERVICE_ACCOUNT_ID"
+  ENVIRONMENT=$(cloud_create_and_use_environment $ENVIRONMENT_NAME)
+  echo "ENVIRONMENT: $ENVIRONMENT, ENVIRONMENT_NAME: $ENVIRONMENT_NAME"
+
+  CLUSTER_NAME=demo-kafka-cluster-$SERVICE_ACCOUNT_ID
+  CLUSTER_CLOUD="${CLUSTER_CLOUD:-aws}"
+  CLUSTER_REGION="${CLUSTER_REGION:-us-west-2}"
+  CLUSTER=$(cloud_create_and_use_cluster $CLUSTER_NAME $CLUSTER_CLOUD $CLUSTER_REGION)
+  BOOTSTRAP_SERVERS=$(ccloud kafka cluster describe $CLUSTER -o json | jq -r ".endpoint" | cut -c 12-)
+  CLUSTER_CREDS=$(cloud_create_credentials_resource $SERVICE_ACCOUNT_ID $CLUSTER)
+  echo "CLUSTER: $CLUSTER, BOOTSTRAP_SERVERS: $BOOTSTRAP_SERVERS, CLUSTER_CREDS: $CLUSTER_CREDS"
+
+  MAX_WAIT=720
+  echo "Waiting for Confluent Cloud cluster to be ready and for credentials to propagate"
+  retry $MAX_WAIT check_ccloud_cluster_ready || exit 1
+  # Estimating another 60s wait still sometimes required
+  echo "Sleeping another 60s"
+  sleep 60
+
+  SCHEMA_REGISTRY_GEO="${SCHEMA_REGISTRY_GEO:-us}"
+  SCHEMA_REGISTRY=$(cloud_enable_schema_registry $CLUSTER_CLOUD $SCHEMA_REGISTRY_GEO)
+  SCHEMA_REGISTRY_ENDPOINT=$(ccloud schema-registry cluster describe -o json | jq -r ".endpoint_url")
+  SCHEMA_REGISTRY_CREDS=$(cloud_create_credentials_resource $SERVICE_ACCOUNT_ID $SCHEMA_REGISTRY)
+  echo "SCHEMA_REGISTRY: $SCHEMA_REGISTRY, SCHEMA_REGISTRY_ENDPOINT: $SCHEMA_REGISTRY_ENDPOINT, SCHEMA_REGISTRY_CREDS: $SCHEMA_REGISTRY_CREDS"
+
+  if $enable_ksql ; then
+    KSQL_NAME="demo-ksql-$SERVICE_ACCOUNT_ID"
+    KSQL=$(cloud_create_ksql_app $KSQL_NAME $CLUSTER)
+    KSQL_ENDPOINT=$(ccloud ksql app describe $KSQL -o json | jq -r ".endpoint")
+    KSQL_CREDS=$(cloud_create_credentials_resource $SERVICE_ACCOUNT_ID $KSQL)
+    echo "KSQL: $KSQL, KSQL_ENDPOINT: $KSQL_ENDPOINT, KSQL_CREDS: $KSQL_CREDS"
+  fi
+
+  cloud_create_wildcard_acls $SERVICE_ACCOUNT_ID
+  ccloud kafka acl list --service-account $SERVICE_ACCOUNT_ID
+
+  CLIENT_CONFIG="/tmp/client-$SERVICE_ACCOUNT_ID.config"
+  cat <<EOF > $CLIENT_CONFIG
+ssl.endpoint.identification.algorithm=https
+sasl.mechanism=PLAIN
+security.protocol=SASL_SSL
+bootstrap.servers=${BOOTSTRAP_SERVERS}
+sasl.jaas.config=org.apache.kafka.common.security.plain.PlainLoginModule required username\="`echo $CLUSTER_CREDS | awk -F: '{print $1}'`" password\="`echo $CLUSTER_CREDS | awk -F: '{print $2}'`";
+basic.auth.credentials.source=USER_INFO
+schema.registry.url=${SCHEMA_REGISTRY_ENDPOINT}
+schema.registry.basic.auth.user.info=`echo $SCHEMA_REGISTRY_CREDS | awk -F: '{print $1}'`:`echo $SCHEMA_REGISTRY_CREDS | awk -F: '{print $2}'`
+ksql.endpoint=${KSQL_ENDPOINT}
+ksql.basic.auth.user.info=`echo $KSQL_CREDS | awk -F: '{print $1}'`:`echo $KSQL_CREDS | awk -F: '{print $2}'`
+EOF
+  echo
+  echo "$CLIENT_CONFIG:"
+  cat $CLIENT_CONFIG
```

Comment:
This results in the config file being printed to the console twice.  IMO remove this one, and let the caller script inform the user of the file contents when "all said and done".

Reply:
Ack and resolved in latest commit